serde_v8/magic/buffer.rs

```diff
@@ -13,6 +13,17 @@ pub struct JsBuffer(V8Slice);
 
 impl_magic!(JsBuffer);
 
+impl JsBuffer {
+  pub fn from_parts(slice: V8Slice) -> Self {
+    Self(slice)
+  }
+
+  /// Returns the buffer to its parts.
+  pub fn into_parts(self) -> V8Slice {
+    self.0
+  }
+}
+
```

Comment:
Could you explain why changes to `serde_v8` are needed?

Reply:
Basically we need to deconstruct JsBuffer and V8Slice to properly return them to V8 forms without cloning. The `to_v8` functions are for serde_v8 and do a lot of extra work when we can literally just break these objects down (since we know we own them) and return them to v8.